django/db/backends/oracle/operations.py

```diff
@@ -29,12 +29,22 @@ class DatabaseOperations(BaseDatabaseOperations):
 DECLARE
     table_value integer;
     seq_value integer;
+    seq_name user_tab_identity_cols.sequence_name%%TYPE;
 BEGIN
+    BEGIN
+        SELECT sequence_name INTO seq_name FROM user_tab_identity_cols
+        WHERE  table_name = '%(table_name)s' AND
+               column_name = '%(column_name)s';
+        EXCEPTION WHEN NO_DATA_FOUND THEN
+            seq_name := '%(no_autofield_sequence_name)s';
+    END;
```

Comment:
With extra `BEGIN/END` we can handle `NO_DATA_FOUND` exception (i.e. new-style sequence doesn't exist) and go forward with the old-style sequence name.